controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -755,6 +761,12 @@ func (r *KubeadmControlPlaneReconciler) reconcileDelete(ctx context.Context, clu
 	for i := range machinesToDelete {
 		m := machinesToDelete[i]
 		logger := logger.WithValues("machine", m)
+		// If etcd leadership is on machine that is about to be deleted, move it to first follower
+		if err := workloadCluster.ForwardEtcdLeadership(ctx, m); err != nil {
```

Comment:
It's not clear to me the advantages of doing this.

We appear to be removing a cluster with this operation.  All the other machines are gone.  We get to this section, we're deleting the remaining control plane machines.  Since we cordon and drain before a machine is deleted, it appears that we'll just assign etcd into a circle: 1->2->3->1 by the time this for loop completes.  It's also not clear why we need to do this at all, since the cluster is going away.

Reply:
Nice catch! The original intention was to do this for control plane scale down operations rather than control plane deletion.

Reply:
@detiber that makes a lot more sense.  I thought it was something like that, but the linked issue was not very specific.

Reply:
👍